vector/src/main/java/im/vector/app/core/pushers/VectorMessagingReceiver.kt

```diff
@@ -0,0 +1,182 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.pushers
+
+import android.content.Context
+import android.content.Intent
+import android.os.Handler
+import android.os.Looper
+import android.widget.Toast
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.ProcessLifecycleOwner
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import org.json.JSONObject
+import im.vector.app.core.di.ActiveSessionHolder
+import im.vector.app.core.extensions.vectorComponent
+import im.vector.app.features.badge.BadgeProxy
+import im.vector.app.features.notifications.NotifiableEventResolver
+import im.vector.app.features.notifications.NotificationDrawerManager
+import im.vector.app.features.notifications.NotificationUtils
+import im.vector.app.features.settings.VectorPreferences
+import org.json.JSONException
+import org.matrix.android.sdk.api.MatrixCallback
+import org.unifiedpush.android.connector.MessagingReceiver
+import timber.log.Timber
+import org.unifiedpush.android.connector.MessagingReceiverHandler
+
+/**
+ * Class extending MessagingReceiverHandler.
+ */
+val upHandler = object: MessagingReceiverHandler {
+
+    private lateinit var notificationDrawerManager: NotificationDrawerManager
+    private lateinit var notifiableEventResolver: NotifiableEventResolver
+    private lateinit var pusherManager: PushersManager
+    private lateinit var activeSessionHolder: ActiveSessionHolder
+    private lateinit var vectorPreferences: VectorPreferences
+
+    fun initVar(context: Context) {
+        with(context.vectorComponent()) {
+            notificationDrawerManager = notificationDrawerManager()
+            notifiableEventResolver = notifiableEventResolver()
+            pusherManager = pusherManager()
+            activeSessionHolder = activeSessionHolder()
+            vectorPreferences = vectorPreferences()
+        }
+    }
+
+    override fun onMessage(context: Context?, message: String, instance: String) {
+        initVar(context!!)
+        Timber.i("onMessage received")
+        lateinit var data: JSONObject
+        lateinit var notification: JSONObject
+        lateinit var eventId: String
+        try {
+            data = JSONObject(message)
+            notification = data.getJSONObject("notification")
+        } catch (e: JSONException) {
+            e.printStackTrace()
+            return
+        }
+        try {
+            eventId = notification.getString("event_id")
+        } catch (e: JSONException) {
+            Timber.i("No event_id on notification")
```

Comment:
Maybe make this (and the other one) `Timber.w`, warnings?

Reply:
Notifications without event_id happen when you read a message to decrease the unread number. It happen quite often and isn't a problem